authorization/role/service/role_management_service.go

```diff
@@ -38,3 +47,96 @@ func (r *RoleManagementServiceImpl) ListByResource(ctx context.Context, resource
 func (r *RoleManagementServiceImpl) ListAvailableRolesByResourceType(ctx context.Context, resourceType string) ([]role.RoleDescriptor, error) {
 	return r.repo.RoleRepository().FindRolesByResourceType(ctx, resourceType)
 }
+
+// assign assigns an identity ( users or organizations or teams or groups ) with a role, for a specific resource
+// IMPORTANT: This is a transactional method, which manages its own transaction/s internally
+func (r *RoleManagementServiceImpl) assign(ctx context.Context, identityID uuid.UUID, resourceID string, roleName string) error {
+
+	err := transaction.Transactional(r.tm, func(tr transaction.TransactionalResources) error {
+
+		rt, err := r.repo.ResourceRepository().Load(ctx, resourceID)
+
+		if err != nil {
+			return err
+		}
+
+		roleRef, err := r.repo.RoleRepository().Lookup(ctx, roleName, rt.ResourceType.Name)
+		if err != nil {
+			return err
+		}
+
+		ir := rolerepo.IdentityRole{
+			ResourceID: resourceID,
+			IdentityID: identityID,
+			RoleID:     roleRef.RoleID,
+		}
+
+		return r.repo.IdentityRoleRepository().Create(ctx, &ir)
+	})
+
+	return err
+}
+
+// Assign assigns an identity ( users or organizations or teams or groups ) with a role, for a specific resource
+// IMPORTANT: This is a transactional method, which manages its own transaction/s internally
+func (r *RoleManagementServiceImpl) Assign(ctx context.Context, assignedBy uuid.UUID, identitiesToBeAssigned []uuid.UUID, resourceID string, roleName string) error {
+	// check if the current user token belongs to a user who has the necessary privileges
+	// for assigning roles to other users.
+
+	permissionService := permservice.NewPermissionService(r.repo)
+	hasScope, err := permissionService.HasScope(ctx, assignedBy, resourceID, authorization.ManageTeamsInSpaceScope)
```

Comment:
Can we add a new scope here, the name of this one doesn't fit the description of the operation.

Reply:
Ah yeah, this has to be `ManageUsersInSpaceScope`